src/terrain/routes.clj

```diff
@@ -191,10 +191,13 @@
                :securityDefinitions security-definitions}})
   (middleware
    [wrap-keyword-params
-    wrap-lcase-params
-    wrap-query-params
     clean-context]
    (context "/terrain" []
      (terrain-routes))))
 
-(def app-wrapper (wrap-context-path-adder app "/terrain"))
+(def app-wrapper
+  (-> app
+      (wrap-query-param-remover "ip-address" #{#"^/terrain/secured/bootstrap$"})
```

Comment:
Any reason why `wrap-query-param-remover` and `wrap-context-path-adder` can't go into the `middleware` wrapper above?

Especially since it seems these middleware wrappers are applied "backwards" here (I think), so that `wrap-*-params` are applied to requests before `wrap-context-path-adder`, then finally `wrap-query-param-remover`?

I like that the order wrappers are listed in the `middleware` wrapper above is the same order they are applied to requests.

Reply:
That's a good point. `context-path-adder` needs to go outside the app definition because of the way that the Swagger UI is handled. The others may be able to go inside it, though. I'll give it a try.

Reply:
I couldn't quite get it to work without modifying the middleware, but a small middleware change was enough to get it to work.